subprojects/core/src/main/java/org/gradle/internal/composite/CompositeBuildSettingsLoader.java

```diff
@@ -45,7 +45,10 @@ public CompositeBuildSettingsLoader(SettingsLoader delegate, ServiceRegistry bui
 
     @Override
     public SettingsInternal findAndLoadSettings(GradleInternal gradle) {
+        CompositeContextBuilder compositeContextBuilder = buildServices.get(CompositeContextBuilder.class);
```

Comment:
Not sure why I didn't do this the first time around: https://github.com/gradle/gradle/commit/2712adbbeb43a0f710017cc514cf900ab89e4d16

There's an odd dependency where the `:core` platform requires an implementation of `CompositeContextBuilder` and `IncludedBuildFactory` at runtime. These interfaces are defined in `:core` but implementations are provided by `:compositeBuilds`. This means that the `:core` project has a test runtime dependency on `:compositeBuilds`. There's probably a better way to structure this.